test/unit/specs/utils/validation/createOrValidator.spec.js

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import {
+  objectOf,
+  boolean,
+  arrayOf,
+  string,
+  literal
+} from "../../../../../src/utils/validation";
+import describeValidation from "./describeValidation";
+
+describe("validation:or", () => {
+  describeValidation(
+    "require one or the other",
+    objectOf({
+      viewStart: boolean(),
+      scopes: arrayOf(string()).nonEmpty()
+    })
+      .required()
+      .or(
+        [
+          objectOf({ viewStart: literal(true).required() }),
+          objectOf({
+            scopes: arrayOf(string())
+              .nonEmpty()
+              .required()
+          })
+        ],
+        "either viewStart set to true or scopes set to a nonEmpty array"
+      ),
```

Comment:
(I already talked to @jonsnyder about this but I'm posting this comment anyway for reference)

This is a pretty interesting approach. Have you thought about how we might support something like: the value is a string or an array? I'm wondering how that would play into the deign or.....maybe we don't worry about it until it comes up.

Maybe something like:

```
oneOf([
  string(),
  arrayOf(string()
])
```

I think that would change your test example to something like:

```
oneOf([
   objectOf({ viewStart: literal(true).required() }),
    objectOf({
       scopes: arrayOf(string())
         .nonEmpty()
         .required()
     })
  ],
  "either viewStart set to true or scopes set to a nonEmpty array"
).required(),
```

Maybe you've thought this through already. I'm not sure how it would affect the validator architecture.

Reply:
this way it looks even better and clear

Reply:
BTW, json-schema has this concept. It might be worth considering how json-schema distinguishes between `anyOf` and `oneOf`. If you applied their terminology to our example here, using the term `anyOf` would probably be most appropriate. https://json-schema.org/understanding-json-schema/reference/combining.html#anyof

Reply:
In case you would like to implement an enum validator while you're at it, I logged a JIRA issue: https://jira.corp.adobe.com/browse/CORE-40687